hack/conformance-test.sh

```diff
@@ -14,26 +14,97 @@
-  echo "Must set KUBE_MASTER_IP before running conformance test."
-  exit 1
-fi
-if [[ -z "AUTH_CONFIG" ]]; then
-  echo "Must set AUTH_CONFIG before running conformance test."
-  exit 1
-fi
+# TODO: when preparing to release a new major or minor version of Kubernetes,
+# then update above commit number, reevaluate the set of e2e tests,
+# update documentation at docs/getting-started-guides/README.md to have
+# a new column for conformance at that new version, and notify
+# community.
+
+# Instructions:
+#  - Setup a Kubernetes cluster with $NUM_MINIONS nodes (defined below).
+#  - Provide a Kubeconfig file whose current context is set to the
+#    cluster to be tested, and with suitable auth setting.
+#  - Specify the location of that kubeconfig with, e.g.:
+#    declare -x KUBECONFIG="$HOME/.kube/config"
+#  - Specify the location of the master with, e.g.:
+#    declare -x KUBE_MASTER_IP="1.2.3.4"
+#  - Make sure only essential pods are running and there are no failed/pending pods.
+#  - Make binaries needed by e2e, e.g.:
+#      make clean
+#      make quick-release
+#  - Run the test and capture output:
+#      hack/conformance-test.sh 2>&1 | tee conformance.$(date +%FT%T%z).log
+
+: ${KUBECONFIG:?"Must set KUBECONFIG before running conformance test."}
+: ${KUBE_MASTER_IP:?"Must set KUBE_MASTER_IP before running conformance test."}
+echo "Conformance test using ${KUBECONFIG} against master at ${KUBE_MASTER_IP}"
+echo -n "Conformance test run date:"
+date
+echo -n "Conformance test SHA:"
+HEAD_SHA=$(git rev-parse HEAD)
+echo $HEAD_SHA
+echo "Conformance test version tag(s):"
+git show-ref | grep $HEAD_SHA | grep refs/tags
+echo
+echo "Conformance test checking conformance with Kubernetes version 1.0"
+
+# It runs a whitelist of tests.  This whitelist was assembled at commit
+# b70b7084c93d4ce80b7463f48c23d5ac04edb2b1 starting from this list of tests:
+#   grep -h 'It(\|Describe(' -R test
+#
+# List of test name patterns not included and why not included:
+#  Cadvisor: impl detail how stats gotten from containers.
+#  MasterCerts: GKE/GCE specific
+#  Density: performance
+#  Cluster level logging...: optional feature
+#  Etcd failure: reliability
+#  Load Capacity: performance
+#  Monitoring: optional feature.
+#  Namespaces.*seconds: performance.
+#  Pod disks: uses GCE specific feature.
+#  Reboot: node management
+#  Nodes: node management.
+#  Restart: node management.
+#  Scale: performance
+#  Services.*load balancer: not all cloud providers have a load balancer.
+#  Services.*NodePort: flaky
+#  Shell: replies on optional ssh access to nodes.
+#  SSH: optional feature.
+#  Volumes: contained only skipped tests.
+export CONFORMANCE_TEST_SKIP_REGEX="Cadvisor|MasterCerts|Density|Cluster\slevel\slogging.*|Etcd\sfailure.*|Load\sCapacity|Monitoring|Namespaces.*seconds|Pod\sdisks|Reboot|Restart|Nodes|Scale|Services.*load\sbalancer|Services.*NodePort|Shell|SSH|Volumes"
+
+declare -x KUBERNETES_CONFORMANCE_TEST="1"
```

Comment:
This isn't meant to imply a version.  
1 is just a string that passes this test: `[[ -n "${KUBERNETES_CONFORMANCE_TEST}" ]]`.